File: paper/src/main/java/com/al3x/housing2/Enums/Locations.java
package com.al3x.housing2.Enums;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.bukkit.Material;

@Getter
@AllArgsConstructor
public enum Locations implements EnumMaterial {
    CUSTOM(Material.COMPASS),
    HOUSE_SPAWN(Material.GRASS_BLOCK),
    PLAYER_LOCATION(Material.LIGHTNING_ROD),
    INVOKERS_LOCATION(Material.PLAYER_HEAD),
    ;

    private final Material material;
    public static Locations fromString(String string) {
        for (Locations location : values()) {
            if (location.name().equalsIgnoreCase(string)) {
                return location;
            }
        }
        return null;
    }
}

File: paper/src/main/java/com/al3x/housing2/Enums/Locations.kt
package com.al3x.housing2.Enums;

public enum Locations {
    CUSTOM,
    HOUSE_SPAWN,
    PLAYER_LOCATION,
    INVOKERS_LOCATION,
    ;

    public static Locations fromString(String string) {
        return switch (string) {
            case "CUSTOM" -> CUSTOM;
            case "HOUSE_SPAWN" -> HOUSE_SPAWN;
            case "PLAYER_LOCATION" -> PLAYER_LOCATION;
            case "INVOKERS_LOCATION" -> INVOKERS_LOCATION;
            default -> null;
        };
    }
}
